### ISSUE TITLE
problem : setting default value and throwing errors case param 

### ISSUE THREAD

**Issue by piya23300** [nicolasblanco/rails_param#18]
1 when param type is boolean but  param value isn't boolean value. it will set param value to nil
I think it should throw error as same as other param type.

example

```
action controller receive `param[:test_boolean] = '123456'`
param! :test_boolean, :boolean
param[:test_boolean] # return nil
```

---

2 when param type is boolean and set default value is boolean

you code is `options[:default].present?` to check default presence
when `default: boolean` ? it's wrong.

example

```
param! :test_boolean, :boolean, default: false
```

`options[:default].present?` will return `false`
this param will not be set to `false` value

**Comment by piya23300** [comment]
This PR #17

**Comment by iMacTia** [comment]
fixed in #17
